### PR TITLE
Feature/external activity tests

### PR DESCRIPTION
**Pull request by senioroman4uk**
Add tests for external activity routes, controller and extensions
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350449%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350582%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350746%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350901%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350954%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63351002%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63351089%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63351205%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63351608%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63352590%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63397828%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63398046%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63398241%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63858632%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20b2a8b4bac268257de1b2a91db0573aafdd7066f3%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350746%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22As%20I%20remember%20we%20already%20agreed%20class%20name%20for%20our%20tests.%20It%20should%20be%20controller%20name%20%2B%20Tests.%20Can%20you%20explain%20what%20means%20GetActivity%20postfix%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A21%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22Understood%20I%27ll%20rename%20it%20as%20it%20has%20to%20be%20one%20test%20per%20controller.%20Understood%20I%20looked%20on%20our%20previous%20tests%2C%20my%20bad%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A28%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10225385%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/senioroman4uk%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs%3AL1-155%22%7D%2C%20%22Pull%20b2a8b4bac268257de1b2a91db0573aafdd7066f3%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs%2055%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350954%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22May%20be%20it%27s%20better%20to%20use%20attribute%20InlineData%28%29%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A23%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs%3AL1-170%22%7D%2C%20%22Pull%20b2a8b4bac268257de1b2a91db0573aafdd7066f3%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63351205%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20are%20mocking%20IGitHubEventsQuery%20and%20then%20call%20GetNumberOfPages%20method%3F%20Are%20you%20sure%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A25%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20see%20then%20it%27s%20ok.%22%2C%20%22created_at%22%3A%20%222016-05-16T18%3A11%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs%3AL1-170%22%7D%2C%20%22Pull%20b2a8b4bac268257de1b2a91db0573aafdd7066f3%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350449%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20it%20correct%20class%20name%20IdentityExtensions%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A19%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22It%20is%20an%20extension%20for%20mocking%20identity%20in%20other%20our%20test%20projects%20it%20is%20called%20this%20way%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A23%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10225385%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/senioroman4uk%22%7D%7D%2C%20%7B%22body%22%3A%20%22And%20it%27s%20extend%20ApiController%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T18%3A10%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%22%2C%20%22created_at%22%3A%20%222016-05-19T10%3A59%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10225385%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/senioroman4uk%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs%3AL1-18%22%7D%2C%20%22Pull%20b2a8b4bac268257de1b2a91db0573aafdd7066f3%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs%2062%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63351002%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22What%20is%20expected%20value%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A23%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22It%20is%20expected%20to%20have%20a%20value%2C%20not%20to%20be%20null.%20I%20wrote%20separate%20test%20that%20checks%20exact%20value%20lower.%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A34%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10225385%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/senioroman4uk%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs%3AL1-170%22%7D%2C%20%22Pull%20b2a8b4bac268257de1b2a91db0573aafdd7066f3%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/157%23discussion_r63350582%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Am%20I%20right%20understand%20that%20I%20can%20apply%20this%20extension%20method%20to%20any%20string%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A20%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%2C%20it%20is.%20It%20looked%20weird%20to%20me%2C%20but%20in%20our%20other%20tests%20it%20is%20made%20this%20way%22%2C%20%22created_at%22%3A%20%222016-05-16T13%3A24%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10225385%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/senioroman4uk%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hmm...Then%20it%27s%20my%20fault%20that%20I%20passed%20such%20extension%20methods.%20Can%20you%20refactor%20this%3F%22%2C%20%22created_at%22%3A%20%222016-05-16T18%3A13%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs%3AL1-13%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull b2a8b4bac268257de1b2a91db0573aafdd7066f3 DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs 7'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/157#discussion_r63350449'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs:L1-18</a></b>
- <a href='https://github.com/TerraVenil'></a> Is it correct class name IdentityExtensions?
- <a href='https://github.com/senioroman4uk'></a> It is an extension for mocking identity in other our test projects it is called this way
- <a href='https://github.com/TerraVenil'></a> And it's extend ApiController?
- <a href='https://github.com/senioroman4uk'></a> Yes
- [ ] <a href='#crh-comment-Pull b2a8b4bac268257de1b2a91db0573aafdd7066f3 DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs 7'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/157#discussion_r63350582'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs:L1-13</a></b>
- <a href='https://github.com/TerraVenil'></a> Am I right understand that I can apply this extension method to any string?
- <a href='https://github.com/senioroman4uk'></a> Yes, it is. It looked weird to me, but in our other tests it is made this way
- <a href='https://github.com/TerraVenil'></a> Hmm...Then it's my fault that I passed such extension methods. Can you refactor this?
- [ ] <a href='#crh-comment-Pull b2a8b4bac268257de1b2a91db0573aafdd7066f3 DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs 20'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/157#discussion_r63350746'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs:L1-155</a></b>
- <a href='https://github.com/TerraVenil'></a> As I remember we already agreed class name for our tests. It should be controller name + Tests. Can you explain what means GetActivity postfix?
- <a href='https://github.com/senioroman4uk'></a> Understood I'll rename it as it has to be one test per controller. Understood I looked on our previous tests, my bad
- [x] <a href='#crh-comment-Pull b2a8b4bac268257de1b2a91db0573aafdd7066f3 DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs 55'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/157#discussion_r63350954'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs:L1-170</a></b>
- <a href='https://github.com/TerraVenil'></a> May be it's better to use attribute InlineData()?
- [ ] <a href='#crh-comment-Pull b2a8b4bac268257de1b2a91db0573aafdd7066f3 DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs 62'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/157#discussion_r63351002'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs:L1-170</a></b>
- <a href='https://github.com/TerraVenil'></a> What is expected value?
- <a href='https://github.com/senioroman4uk'></a> It is expected to have a value, not to be null. I wrote separate test that checks exact value lower.
- [x] <a href='#crh-comment-Pull b2a8b4bac268257de1b2a91db0573aafdd7066f3 DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs 45'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/157#discussion_r63351205'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs:L1-170</a></b>
- <a href='https://github.com/TerraVenil'></a> You are mocking IGitHubEventsQuery and then call GetNumberOfPages method? Are you sure?
- <a href='https://github.com/TerraVenil'></a> I see then it's ok.

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/157?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/157?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-013/pull/157'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ITsvetkoFF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ITsvetkoFF/Kv-013/pull/157'>Refresh</a>.<a href='#crh-update-none'></a>

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Security.Claims;
+using System.Security.Principal;
+using System.Web.Http;
+
+namespace GitHubExtension.Activity.External.Tests.Extensions
+{
+    public static class IdentityExtensions
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs:7]: Is it correct class name IdentityExtensions?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using GitHubExtension.Activity.External.WebAPI;
+
+namespace GitHubExtension.Activity.External.Tests.TestsForRoutes.Extensions
+{
+    public static class ExternalActivityRouteTestExtensions
+    {
+        public static string ForGetGitHubActivityRoute(this string url, int page = 1)
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs]: Am I right understand that I can apply this extension method to any string object?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+using GitHubExtension.Activity.External.WebAPI.Models;
+using GitHubExtension.Activity.External.WebAPI.Models.EventPayloads;
+using GitHubExtension.Activity.External.WebAPI.Queries;
+using GitHubExtension.Infrastructure.Constants;
+using NSubstitute;
+using Xunit;
+
+namespace GitHubExtension.Activity.External.Tests.TestsForControllers
+{
+    public class ActivityControllerGetActivityTests
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs]: As I remember we already agreed class name for our tests. It should be controller name + Tests. Can you explain what means GetActivity postfix?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs (hunk 1 above)
Comment (senioroman4uk) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs:7]: It is an extension for mocking identity in other our test projects it is called this way

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        
+            // Assert
+            numberOfPages.Should().NotHaveValue("link header can not be present if we are on a last page or there is only one page");
+        }
+
+        [Fact]
+        public void ShouldNotReturnNullWhenLinkHeaderIsNotNull()
+        {
+            // Arrange
+            IEnumerable<string> requestLinkHeader = new List<string>() { HeaderForShouldNotReturnNullWhenLinkHeaderIsNotNull };
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs]: May be it's better to use attribute InlineData()?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+            IGitHubEventsQuery query = Substitute.For<IGitHubEventsQuery>();
+
+            // Act
+            int? numberOfPages = query.GetNumberOfPages(requestLinkHeader);
+
+            // Assert
+            numberOfPages.Should().HaveValue();
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs]: What is expected value?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs (hunk 2 above)
Comment (senioroman4uk) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs]: Yes, it is. It looked weird to me, but in our other tests it is made this way

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            
+        [Fact]
+        public void ShouldReturnNullWhenLinkHeaderIsNull()
+        {
+            // Arrange
+            IEnumerable<string> requestLinkHeader = null;
+            IGitHubEventsQuery query = Substitute.For<IGitHubEventsQuery>();
+
+            // Act
+            int? numberOfPages = query.GetNumberOfPages(requestLinkHeader);
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs]: You are mocking IGitHubEventsQuery and then call GetNumberOfPages method? Are you sure?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs (hunk 3 above)
Comment (senioroman4uk) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForControllers/ActivityControllerGetActivityTests.cs]: Understood I'll rename it as it has to be one test per controller. Understood I looked on our previous tests, my bad

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs (hunk 6 above)
Comment (senioroman4uk) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs]: Get number of pages it is just parsing a string, So, I mocked my query to use extension on mock

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs (hunk 5 above)
Comment (senioroman4uk) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs]: It is expected to have a value, not to be null. I wrote separate test that checks exact value lower.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs (hunk 1 above)
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs:7]: And it's extend ApiController?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs (hunk 6 above)
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs]: I see then it's ok.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs (hunk 2 above)
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForRoutes/Extensions/ExternalActivityRouteTestExtensions.cs]: Hmm...Then it's my fault that I passed such extension methods. Can you refactor this?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs (hunk 1 above)
Comment (senioroman4uk) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/Extensions/IdentityExtensions.cs:7]: Yes

**Pull request merged**